eth/filters/api.go

```diff
@@ -280,8 +321,175 @@ func (api *FilterAPI) Logs(ctx context.Context, crit FilterCriteria) (*rpc.Subsc
 			}
 		}
 	}()
+	return nil
+}
 
-	return rpcSub, nil
+// histLogs retrieves logs older than current header.
+func (api *FilterAPI) histLogs(notifier notifier, rpcSub *rpc.Subscription, from int64, crit FilterCriteria) error {
```

Comment:
I think `histLogs` is a bit wonky, is the benefit of reducing `historicalLogs` really worth it?